client/cd-create-profile.c

```diff
@@ -681,6 +685,52 @@ cd_util_icc_set_metadata_coverage (CdIcc *icc, GError **error)
 	return TRUE;
 }
 
+static gboolean
+cd_util_adjust_creation_time (CdUtilPrivate *priv, const char *source_filename, GError **error)
+{
+	GStatBuf st;
+	g_autoptr (GDateTime) creation_time;
+	gint64 build_date = 0;
+	char *source_date_epoch_string;
+	gint64 source_date_epoch;
+	char *endptr;
+
+	/* adjust creation time to match the XML one or SOURCE_DATE_EPOCH if specified and older */
+	source_date_epoch_string = getenv ("SOURCE_DATE_EPOCH");
+	if (source_date_epoch_string) {
+		errno = 0;
+		source_date_epoch = strtoull (source_date_epoch_string, &endptr, 10);
```

Comment:
eewww, can we do this with anything more modern? i.e. GDateTime?

Reply:
Surprisingly it seems not.

There doesn't seem to be any GDateTime-from-string, and the `g_ascii_strtoull` documentation suggests you should probably be using `strtoull` instead…